draft-pda-protocol.md

```diff
@@ -573,116 +612,288 @@ PA protocol. The structure of the leader share is similarly protocol specific:
+
+[TODO: Decide if and how the collector's request is authenticated.]
+
+The collector interacts with the leader to produce the final aggregate output.
+This process consists of a sequence of *collect requests* issued to the leader.
+Before a request can succeed, the aggregators must have verified and aggregated
+enough reports and the leader must have obtained the helper's encrypted output
+share (see {{pa-aggregate}}). In general, the procedure by which the aggregators
+verify and aggregate reports depends on parameters carried by the collect
+request. This procedure is described below in {{pa-aggregate}}.
+
+### Collect Request
+
+The collect protocol is an iterative procedure driven by the collector and
+parameterized by a PAParam. At a high level, it proceeds as follows. First, the
+collector initializes local per-protocol state using the corresponding
+PAParam. This state object has the following member functions:
+
+- CreateRequest(): Creates a PACollectReq object, defined below, that is sent to
+  the leader to complete one iteration of the protocol. Along with any
+  protocol-specific parameters, the message specifies a time interval
+  `[batch_start, batch_end)` that determines the batch of reports to be
+  aggregated. It must be that `batch_end - batch_start >= PAParam.batch_window`
+  and `batch_start` and `batch_end` are multiples of `PAParam.batch_window`.
+
+~~~
+struct {
+  PATask task;
+  Url helper;
+  uint64 batch_start; // The beginning of the batch in UNIX time.
+  Uint64 batch_end;   // The end of the batch in UNIX time (exclusive).
+  PAProto proto;
+  select (PACollectReq.proto) {
+    case prio: PrioCollectReq;
+    case hits: HitsCollectReq;
+  }
+} PACollectReq;
+~~~
 
-## Verify {#pa-verify}
+- Update(resp: PACollectResp): Consumes an opaque PACollectResp object, defined
+  below, that is received from the leader in response to a PACollectReq.
+
+~~~
+struct {
+  PATask task;
+  PAProto proto;
+  PAOutputShare leader_share;
+  opaque encrypted_helper_share;
+} PACollectResp;
+~~~
+
+- Finished(): Returns a boolean indicating indicating whether or not the collection
+  procedure is complete. This function returns true when the collect iteration
+  is complete.
+- Output(): Returns the aggregate output corresponding to the protocol.
+
+The collect procedure for a given PAParam structure `param` is then driven with
+the following algorithm:
+
+~~~
+state = CreateState(param)
+while not state.Finished():
+   req = state.CreateRequest()
+   resp = fetch([leader]/collect, req)
+   state.Update(resp)
+return state.Output()
+~~~
 
-[TODO: Add an illustration of this sub-protocol.]
+[OPEN ISSUE: Describe how intra-protocol errors yield collect errors (see
+issue#57). For example, how does a leader respond to a collect request if the
+helper drops out?]
+
+Each collect request involves one of the helpers specified by the PA parameters.
+If more than one helper is specified, the collector may issue the requests in
+any order.
```

Comment:
As noted in in issue #68 this just seems like extra complexity. The helpers for any given task should be fixed.

Reply:
Ack, I'm sending a PR today to rip this out.